lib/streamlit/runtime/caching/hashing.py

```diff
@@ -405,7 +405,7 @@ def _to_bytes(self, obj: Any) -> bytes:
         elif obj is False:
             return b"0"
 
-        elif dataclasses.is_dataclass(obj):
+        elif dataclasses.is_dataclass(obj) and not isinstance(obj, type):
```

Comment:
Based on this comment from a mypy issue thread: https://github.com/python/mypy/issues/17550#issuecomment-2242651365